### ISSUE TITLE
[PSR] rename EightPointsGuzzleBundlePlugin

### ISSUE THREAD

**Issue by gregurco** [8p/EightPointsGuzzleBundle#178]
Rename EightPointsGuzzleBundlePlugin to EightPointsGuzzleBundlePlugin**Interface**. This should be done in v8, not to break BC.

**Comment by rrajkomar** [comment]
Sorry I'm using this topic because I have a question related to the whole plugin system.
Why use bundle class to create plugins ? Why not simply define tags for different types of elements (middleware, event listeners) ?
Since you provide a base interface any class that implements it can be used as a plugin element given a tagged service is created for the class. 

The tag would then be used to load the plugin into the list of plugins.
Also by providing the list of guzzle clients for which to use the plugin in the tag or by defining one tag per guzzle client (I prefer the list system as it requires less lines in services config), you can configure for which client each plugin is used.

I know this would be a lot to change but you talking here about v8 so bc might not be an issue.
What's your take on all this ?

**Comment by gregurco** [comment]
Hello. This decision is not the random one. I spend days and days to find any other solution and I can write just one hint: configuration tree of bundle is built in the moment when container, tags, services, autowire and so on are not existing. So, it's the level where you don't have this symfony "sugar" :slightly_frowning_face:

**Comment by gregurco** [comment]
Compiler Passes are executed later and out of the bundle scope. So in Compiler Pass you can define services and so on, but you can not modify the configuration tree of specific bundle.

**Comment by rrajkomar** [comment]
Yes but the solution I was suggesting does not require changing the Configuration tree at that point.
The idea is to dynamically add the element to the HandlerStack and/or the client when executing the Compiler Pass.

The "main" bundle would be looking specifically tagged services which can be defined in other bundles.

**Comment by rrajkomar** [comment]
I'll work on a demo on what I had in mind. In the meantime, could you tell me how to fix the Stopwatch issue when running tests. It is quite cumbersome to test for issues :)

**Comment by gregurco** [comment]
Could you please open separate issue with more details? Please post console output, version of PHP and PhpUnit.
